providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/FlagdProvider.java

```diff
@@ -109,22 +115,80 @@ public FlagdProvider(final FlagdOptions options) {
         this.maxEventStreamRetries = maxEventStreamRetries;
         this.cache = new FlagdCache(cache, maxCacheSize);
         this.eventStreamAliveSync = new Object();
-        this.handleEvents();
     }
 
     @Override
-    public void restartEventStream() throws Exception {
+    public void initialize(EvaluationContext evaluationContext) throws RuntimeException {
+        try {
+            // try a dummy request
+            this.serviceBlockingStub
+                    .withWaitForReady()
+                    .withDeadlineAfter(this.deadline, TimeUnit.MILLISECONDS)
+                    .resolveBoolean(ResolveBooleanRequest.newBuilder().setFlagKey("ready?").build());
+        } catch (StatusRuntimeException e) {
+            // only return the exception if we don't meet the deadline
+            if (Status.DEADLINE_EXCEEDED.equals(e.getStatus())) {
+                throw e;
+            }
+        } finally {
+            // try in background to open the event stream
+            this.handleEvents();
```

Comment:
I suppose this works, but I think that the main purpose of `initialize` is to connect the stream. I imagine there could be other reasons the stream could fail besides the connection? Perhaps due to some network/fw restrictions? I think the safest way to test the initialize would be to actually open the stream. Is doing that in a blocking way not possible?

I could be misunderstanding something here. Let me know!

Reply:
gRPC builds on top of channels, if there is no connection with the channel, the RPC won't be scheduled/executed. Since both stubs use the same channel, I used the blocking one to perform an evaluation which ensures that we can connect to FlagD. I also preferred using the blocking stub because this is the critical path for communication with FlagD since FF evaluations run over it. The nonblocking stub is used to remove keys from the cache. Do you prefer to check that both of them are set up correctly? 

> Perhaps due to some network/fw restrictions?

do you mean if there is a level 7 firewall? I think running the evaluation would cover this issue since we transmit data over the wire.

Reply:
> do you mean if there is a level 7 firewall? I think running the evaluation would cover this issue since we transmit data over the wire.

You're probably right. TBH I don't feel too strongly about this. As long as we init properly I think that's fine.